File: source-connector/src/main/java/sourceconnector/repository/file/decompressor/ZstdDecompressor.java
package sourceconnector.repository.file.decompressor;

import com.github.luben.zstd.ZstdInputStream;
import offsetmanager.domain.file.FileKey;

import java.io.IOException;
import java.io.InputStream;

public class ZstdDecompressor implements Decompressor {
  @Override
  public boolean supports(FileKey fileKey) {
    return fileKey.get().endsWith(".zst");
  }

  @Override
  public InputStream decompress(InputStream in) throws IOException {
    return new ZstdInputStream(in);
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/decompressor/ZstdDecompressor.kt
package sourceconnector.repository.file.decompressor

import com.github.luben.zstd.ZstdInputStream
import offsetmanager.domain.file.FileKey
import java.io.IOException
import java.io.InputStream

class ZstdDecompressor : Decompressor {
  override fun supports(fileKey: FileKey): Boolean {
    return fileKey.get().endsWith(".zst")
  }

  @Throws(IOException::class)
  override fun decompress(`in`: InputStream): InputStream {
    return ZstdInputStream(`in`)
  }
}
